app/src/main/java/org/kinecosystem/kinit/view/TabsAdapter.kt

```diff
@@ -100,14 +107,34 @@ class TabsAdapter :
             R.layout.pre_earn_tab, parent, false)
         binding.model = PreEarnViewModel()
         models[position] = binding.model
-        moveToEarnTab(parent, position, binding)
+        moveToEarnTab(parent)
         return binding.root
     }
 
-    private fun moveToEarnTab(parent: ViewGroup, position: Int, binding: PreEarnTabBinding) {
+    private fun moveToEarnTab(parent: ViewGroup) {
         if (!preAnimationWasShown && tasksRepository.isTaskAvailable()) {
             parent.postDelayed({
                 if (parent is LockableViewPager) {
+                    parent.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
```

Comment:
I think it would be cleaner to move the code that figures out when the transition is completed inside the LockableViewPager and overload the LockableViewPager setCurrentItem method so that here, you simply do:
```kotlin
parrent.setCurrentItem(EARN_TAB_INDEX, true, object: TransitionCompletedListener {
   override fun onTransitionCompleted(){
      pageSelectionListener?.enablePageSelection()
   } 
}
```